projects/extension/src/background/index.ts

```diff
@@ -33,22 +35,53 @@ declare let window: Background
 window.manager = publicManager
 
 const l = logger("Extension")
+
 export interface RequestRpcSend {
   method: string
   params: unknown[]
 }
 
+const saveChainDbContent = async (key: string, chain: Chain) => {
+  const db = await chain.databaseContent(
+    chrome.storage.local.QUOTA_BYTES / wellKnownChains.size,
+  )
+  chrome.storage.local.set({ [key]: db })
+}
+
+const flushDatabases = (): void => {
+  for (const [key, chain] of wellKnownConnections)
+    saveChainDbContent(key, chain)
+}
+
+chrome.alarms.onAlarm.addListener((alarm) => {
+  if (alarm.name === "DatabaseContentAlarm") flushDatabases()
+})
+
 const init = async () => {
   try {
     manager = new ConnectionManager(start({ maxLogLevel: 3 }))
     for (const [key, value] of wellKnownChains.entries()) {
       const rpcCallback = (rpc: string) => {
         console.warn(`Got RPC from ${key} dummy chain: ${rpc}`)
       }
-      await manager
-        .addChain(value, rpcCallback)
-        .catch((err) => l.error("Error", err))
+
+      const dbContent = await new Promise<string | undefined>((res) =>
+        chrome.storage.local.get([key], (val) => res(val[key] as string)),
+      )
+
+      const chain = await manager.addChain(
+        value,
+        rpcCallback,
+        undefined,
+        dbContent,
+      )
+      wellKnownConnections.set(key, chain)
+      !dbContent && saveChainDbContent(key, chain)
```

Comment:
```suggestion
      if (!dbContent) saveChainDbContent(key, chain)
```

I know `&&` looks cool and all, but the goal is to have easy to read code